crates/sui-adapter/src/adapter.rs

```diff
@@ -336,28 +341,39 @@ pub fn publish<
         + ChildObjectResolver,
 >(
     state_view: &mut S,
-    vm: &MoveVM,
+    _vm: &MoveVM,
```

Comment:
I'm a little bit confused by what this change implies for how things used to work before.  What is the VM that gets passed into this function, and how does it differ from the `vm` that's created by `verify_and_link`?